test/tools/Modules/WebListener/WebListener.psm1

```diff
@@ -60,12 +60,20 @@ function Start-WebListener
         $serverPfx           = 'ServerCert.pfx'
         $serverPfxPassword   = 'password'
         $initCompleteMessage = 'Now listening on'
+        $sleepMilliseconds   = 100
 
         $serverPfxPath = Join-Path $MyInvocation.MyCommand.Module.ModuleBase $serverPfx
-        $timeOut = (get-date).AddSeconds($initTimeoutSeconds)
         $Job = Start-Job {
-            $path = Split-Path -parent (get-command WebListener).Path
-            Push-Location $path
+            $path = Split-Path -parent (get-command WebListener).Path -Verbose
+            Push-Location $path -Verbose
+            'appDLL: {0}' -f $using:appDll
+            'serverPfxPath: {0}' -f $using:serverPfxPath
+            'serverPfxPassword: {0}' -f $using:serverPfxPassword
+            'HttpPort: {0}' -f $using:HttpPort
+            'Https: {0}' -f $using:HttpsPort
+            'Tls11Port: {0}' -f $using:Tls11Port
+            'TlsPort: {0}' -f $using:TlsPort
+            $env:ASPNETCORE_ENVIRONMENT = 'Development'
             dotnet $using:appDll $using:serverPfxPath $using:serverPfxPassword $using:HttpPort $using:HttpsPort $using:Tls11Port $using:TlsPort
```

Comment:
Can `dotnet run` fail here? Should we check a return code or the job return it automatically? Do we get this in  `$initStatus` - if so we should change logic below for ending loop.

Reply:
I don't see any benefit in changing the current logic. the timeout is 15 seconds. Failing early shaves off maybe a few seconds but adds complexity that could result in false fails or false successes. I say we leave it alone.

Reply:
Well, my understanding is that if we don't get `$initCompleteMessage` we in any case get timeout error and a native error code which led to the lack of `$initCompleteMessage`.

I suggest checking this out. Could you please push a temporary commit to emulate an error in WebListener startup?

Reply:
@iSazonov No. we were seeing no output at all when this happens which was indicating that dotent was either not running at all or was running and hung without any output. Even when it fails there is stdout output so we should have seen at least something. instead, we were simply timing out.

Reply:
@iSazonov I have pushed a commit which throws on WebListener start up.

Reply:
Thanks! 
I don't see our message "'WebListener did not start before the timeout was reached ..." in logs.

Reply:
yea.. erm.. `-Force` is not present on `Stop-Job` so that syntax error was popping up instead as it was hit first, so the `throw` message never made it through. I have corrected and repushed. I also had to change the temporary throw WebListener.. it wasn't compiling due to the unreachable code...

Reply:
https://ci.appveyor.com/project/PowerShell/powershell/build/v6.1.0-preview.7742#L2508

```none
Describing Invoke-WebRequest tests
 [-] Error occurred in Describe block 16.43s
   WebListener did not start before the timeout was reached.
   Errors:
   NotSpecified: (:String) [], RemoteException
   Unhandled Exception: System.ApplicationException: Bad things happened.
      at mvc.Program.Main(String[] args) in C:\projects\powershell\test\tools\WebListener\Program.cs:line 23
   
   Output:
   appDLL: WebListener.dll
   serverPfxPath: C:\projects\powershell\test\tools\Modules\WebListener\ServerCert.pfx
   serverPfxPassword: password
   HttpPort: 8083
   Https: 8084
   Tls11Port: 8085
   TlsPort: 8086
   
   Verbose:
   
   At C:\projects\powershell\test\tools\Modules\WebListener\WebListener.psm1:107 char:13
```

Reply:
Opened #6020 to track the issue where one of the describe blocks is missing `Start-WebListener`

Reply:
Looks good!